apacheconfig/wloader.py

```diff
@@ -0,0 +1,204 @@
+#
+# This file is part of apacheconfig software.
+#
+# Copyright (c) 2018-2019, Ilya Etingof <etingof@gmail.com>
+# License: https://github.com/etingof/apacheconfig/LICENSE.rst
+#
+import abc
+import six
+
+from apacheconfig import make_parser
+from apacheconfig import make_lexer
+
+
+def _restore_original(word):
+    """If the `word` is a Quoted string, restores it to original.
+    """
+    if getattr(word, 'is_single_quoted', False):
+        return "'%s'" % word
+    if getattr(word, 'is_double_quoted', False):
+        return '"%s"' % word
+    return word
+
+
+@six.add_metaclass(abc.ABCMeta)
+class Node(object):
+    """Generic class containing data that represents a node in the config AST.
+    """
+
+    @abc.abstractmethod
+    def dump(self):
+        """Returns the contents of this node as it would appear in a config
+        file.
+        """
+
+    @abc.abstractproperty
+    def ast_node_type(self):
+        """Returns object typestring as defined by the apacheconfig parser.
+        """
+
+    @abc.abstractproperty
+    def whitespace(self):
+        """Returns preceding or trailing whitespace for this node as a string.
+       """
```

Comment:
nit: how flake8 overlooked wrong indentation?

Reply:
o.O i'm not sure!